### PR TITLE
Add workflow to resolve unintended git diffs before PR review

### LINKED ISSUE #45: Setup CI to avoid accumulation of small problems
## Task description

- Add GitHub Action workflow to execute CI on pull request

## Background

- codes in this repo are formatted using `@biomejs/biome check --write`, but the execution timing is arbitrary
- I found unintended git diff on local after executing `@biomejs/biome check --write`
- we want to free our contributors from this kind of toils

## TODO (Optional)

- [x] Add `.github/workflows/ci.yml`
- [x] Configure `ci.yml` to eliminate unintended diff before PR review

### PR DESCRIPTION
**Pull request by Rindrics**
## Summary

- Avoid accumulation of small problems by setting up CI

## Related Issue

- close #45 

## Changes

- added GitHub Action workflow to detect uncomitted diff

### the workflow detected bad diffs on codebase
- https://github.com/route06inc/giselle/actions/runs/11492891991
- https://github.com/route06inc/giselle/actions/runs/11492965058
- https://github.com/route06inc/giselle/actions/runs/11493012862

### the workflow succeeded when codebase become healthy

https://github.com/route06inc/giselle/actions/runs/11493035334

### CODE REVIEW

**Review by shige — APPROVED** [review]
LGTM 👍

**Comment by rindrics** [comment]
thx!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Bun
+        uses: oven-sh/setup-bun@v2
+
+      - name: Install dependencies
+        run: bun install
```
Comment (masutaka) [inline .github/workflows/ci.yml:17]: @Rindrics It is better to use the bun version in package.json.

https://github.com/route06inc/giselle/blob/930deb6b9306a2627c0907005c4ed58b1cb52f88/.github/workflows/license.yml#L27-L30

https://github.com/route06inc/giselle/blob/930deb6b9306a2627c0907005c4ed58b1cb52f88/package.json#L14

Comment (masutaka) [reply .github/workflows/ci.yml:17]: I checked that it has been replaced by https://github.com/route06inc/giselle/pull/47

**Pull request merged**